README.md

```diff
@@ -1,81 +1,148 @@
+	    lifecycle to ensure your program stays loaded and that you can easily
+	    tolerate pod restarts and upgrades.
+
+## What is bpfd?
+
+bpfd is a software stack that aims to make it easy to load, unload, modify and
+monitor BPF programs whether on a single host, or in a Kubernetes cluster. bpfd comprises:
+
+  - bpfd: A system daemon that supports loading, unloading, modifying and
+	  monitoring of BPF programs exposed over a gRPC API.
+  - eBPF CRDS: That provide a way to express intent to load BPF programs
+	  `XdpProgram`, `TcProgram`, etc. and represent the runtime state of loaded
+	  programs via `BpfProgram`. 
+  - bpfd-agent: That ensures that the `BpfPrograms` for a given node are in the
+	  desired state.
+  - bpfd-operator: An operator, built using [Operator
+	  SDK](https://sdk.operatorframework.io/) that manages the installation and
+	  lifecycle of bpfd-agent and the CRDs in a Kubernetes cluster.
+
+bpfd is developed in Rust and built on top of Aya, a Rust BPF library.
+
+The benefits of this solution include the following:
+
+- Security
+  - Improved security because only the bpfd daemon, which can be tightly
+    controlled, has the privileges needed to load BPF programs, while access to
+    the API can be controlled via standard RBAC methods.  Within bpfd, only a
+    single thread keeps these capabilities while the other threads (serving
+    RPCs) do not.
+  - Gives the administrators control over who can load programs.
+  - Allows administrators to define rules for the ordering of networking BPF
+    programs. (ROAD MAP)
+- Visibility/Debuggability
+  - Improved visibility into what BPF programs are running on a system, which
+    enhances the debuggability for developers, administrators, and customer
+    support.
+  - The greatest benefit is achieved when all apps use bpfd, but even if they
+    don't, bpfd can provide visibility into all the BPF programs loaded on the
+    nodes in a cluster.
+- Multi-program Support
+  - Support for the coexistence of multiple BPF programs from multiple users.
+  - Uses the [libxdp multiprog
+    protocol](https://github.com/xdp-project/xdp-tools/blob/master/lib/libxdp/protocol.org)
+    to allow multiple XDP programs on single interface
+  - This same protocol is also supported for TC programs to provide a common
+    multi-program user experience across both TC and XDP.
+  - An additional benefit for TC is that it makes the re-ordering and inserting
+    of new programs atomic.
+- Productivity
+  - Simplifies the deployment of BPF programs in a Kubernetes cluster.
+  - Improved ease of use and productivity because developers can use a standard
+    framework for managing BPF programs rather than developing their own.
+  - Developers can still use Cilium/libbpf/Aya/etc libraries for BPF
+    development, and load/unload with bpfd.
+  - Provides BPF Bytecode Image Specifications that allows fine-grained separate
```

Comment:
Ahh cool You had this here :)

You can ignore my comment above if you want

Reply:
I think it's okay -- the first comment describes the problem, while this one describes the solution.